analyzers/src/SonarAnalyzer.CSharp/Rules/StringConcatenationInLoop.cs

```diff
@@ -21,55 +21,26 @@
     {
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
-        protected override bool IsExpressionConcatenation(BinaryExpressionSyntax addExpression) =>
-            addExpression.IsKind(SyntaxKind.AddExpression);
+        protected override SyntaxKind[] CompoundAssignmentKinds { get; } = new[] { SyntaxKind.AddAssignmentExpression };
 
-        protected override SyntaxNode GetLeft(AssignmentExpressionSyntax assignment) => assignment.Left;
-
-        protected override SyntaxNode GetRight(AssignmentExpressionSyntax assignment) => assignment.Right;
-
-        protected override SyntaxNode GetLeft(BinaryExpressionSyntax binary) => binary.Left;
-
-        protected override bool IsInLoop(SyntaxNode node) => LoopKinds.Contains(node.Kind());
-
-        protected override bool AreEquivalent(SyntaxNode node1, SyntaxNode node2) =>
-            CSharpEquivalenceChecker.AreEquivalent(node1, node2);
+        protected override ISet<SyntaxKind> ExpressionConcatenationKinds { get; } = new HashSet<SyntaxKind>
+        {
+            SyntaxKind.AddExpression
+        };
 
-        private static readonly ISet<SyntaxKind> LoopKinds = new HashSet<SyntaxKind>
+        protected override ISet<SyntaxKind> LoopKinds { get; } = new HashSet<SyntaxKind>
         {
             SyntaxKind.WhileStatement,
             SyntaxKind.DoStatement,
             SyntaxKind.ForStatement,
             SyntaxKind.ForEachStatement
         };
 
-        private static readonly ISet<SyntaxKind> AddOperators = new HashSet<SyntaxKind>
-        {
-            SyntaxKind.PlusToken,
-            SyntaxKind.PlusEqualsToken
-        };
-
-        private static readonly ImmutableArray<SyntaxKind> simpleAssignmentKinds =
-            ImmutableArray.Create(SyntaxKind.SimpleAssignmentExpression);
-
-        private static readonly ImmutableArray<SyntaxKind> compoundAssignmentKinds =
-            ImmutableArray.Create(SyntaxKind.AddAssignmentExpression);
-
-        protected override ImmutableArray<SyntaxKind> SimpleAssignmentKinds => simpleAssignmentKinds;
-
-        protected override ImmutableArray<SyntaxKind> CompoundAssignmentKinds => compoundAssignmentKinds;
-
-        protected override Helpers.GeneratedCodeRecognizer GeneratedCodeRecognizer => CSharpGeneratedCodeRecognizer.Instance;
-
-        protected override bool IsAddExpression(BinaryExpressionSyntax rightExpression) =>
-            rightExpression != null &&
-            rightExpression.OperatorToken.IsAnyKind(AddOperators);
+        protected override bool IsAddExpression(BinaryExpressionSyntax expression) =>
+            expression.OperatorToken.IsAnyKind(SyntaxKind.PlusToken, SyntaxKind.PlusEqualsToken);
```

Comment:
* SyntaxKind.PlusEqualsToken is associated with AssignmentExpressionSyntax which is not a BinaryExpression
* Use expression.Kind instead of the operator token kind.

```suggestion
            expression.IsKind(SyntaxKind.AddExpression);
```